tests/test_shell.py

```diff
@@ -247,13 +248,20 @@ def test_find(self):
 
         # Test the doc_filter
         for job in project.find_jobs():
-            job.document['a'] = job.statepoint()['a']
+            job.document['a'] = job.statepoint()['a'] + 1
+
         with pytest.deprecated_call():
-            for i in range(3):
+            for i in range(1, 4):
```

Comment:
What is the motivation for this change?

Reply:
I have changed the value `a` in doc and sp so that they are different for a particular job. So to test the key 'a' selected belong to sp or doc.

Reply:
I'd prefer to just add a new test that looks at that specific behavior. Whenever we modify existing tests, we increase the risk of introducing unintended backwards incompatible changes. It is absolutely ok if we largely duplicate existing tests. Duplication of test code is much preferable over modifying existing tests.

Reply:
Okay, Got it 👍